mteb/evaluation/evaluators/RerankingEvaluator.py

```diff
@@ -71,17 +71,22 @@ def compute_metrics_batched(self, model):
         all_mrr_scores = []
         all_ap_scores = []
 
+        # using encode_queries and encode_corpus functions if they exists,
+        # which can be defined by users to add different instructions for query and passage conveniently
+        encode_queries_func = model.encode_queries if hasattr(model, 'encode_queries') else model.encode
+        encode_corpus_func = model.encode_corpus if hasattr(model, 'encode_corpus') else model.encode
```

Comment:
Nice idea! I think we should add this liberty to all the difference evaluations which would like to have different encodings for different samples.

Can we please do the same change for the other method below: `compute_metrics_individual()`

Reply:
Very sorry for the late reply. 
I think it's a good idea to change the compute_metrics_individual too.

Reply:
Did it already 👍